website/docs/variants.md

```diff
@@ -1,5 +1,129 @@
+
+## Getting started
+
+```tsx
+import { variant } from 'kodiak-ui'
+import { Button } from '@kodiak-ui/primitives'
+
+variant('seconday', {
+  bg: 'white',
+  border: '1px solid',
+  borderColor: 'black',
+  color: 'black',
+  '&:hover': {
+    bg: 'black',
+    color: 'white',
+  },
+})
+
+variant('lg', {
+  px: 4,
+  py: 3,
+})
+
+function App() {
+  return (
+    <>
+      <Button>Click me!</Button>
+      <Button variants={['secondary', 'lg']}>Maybe click me</Button>
+    </>
+  )
+}
+```
+
+As you can see, defining a new variant is as easy as calling `variant` with a key and style object. The style object is theme aware, so you can use all of the goodness that Kodiak UI provides with scaled theme properties for spacing, fonts, radii, etc...
+
+Each variant is stored in a global store that can be accessed from any component in your application no matter where `variant` is called. We recommend that you think about variants in a similar way to approaching state. Typically state is a top -> down approach and co-located to where the state is used. Variants should be defined in the same way. Define your variants initially in a top down approach and co-locate them to where they are going to be used. If you need to use them in another area of your application, move the variant definitions up the tree.
+
+If you do need to utilize a variant in a parent that is defined in a child component, the `useVariant` hook can be applied.
+
+```tsx
+import { useVariant } from 'kodiak-ui'
+import { secondaryVariant } from './some-other-file'
+
+function Parent() {
+  const secondary = useVariant(secondaryVariant)
+
+  return <Button sx={{ ...secondary }} />
+}
+```
+
+## API
+
+Kodiak UI's main ways of working with variants.
+
+### variant
+
+The `variant` method allows you to define variants that are then injected into your theme from anywhere in your application.
+
+```ts
+variant('key', { bg: 'primary', px: 4 })
+```
+
+| Argument | Type   | Description                                                                          |
+| -------- | ------ | ------------------------------------------------------------------------------------ |
+| key      | string | The name of the variant and how you will reference the variant in the component prop |
+| styles   | object | Theme scoped style object. The styles that will be applied to the component          |
+
+💡**Tip**: See the alternate way to define variants [below](#defining-variants-globally).
+
+### variants prop
+
+All Kodiak UI components accept a `variants` prop which will take a string or array of strings as a value. The variant(s) passed into the `variants` prop will apply the appropriate styling to the component.
+
+```tsx
+<Button variants="secondary">Click me!</Button>
+<Button variants={['secondary', 'lg']}>Click me!</Button>
```

Comment:
This is great!  I just realized that emotion allows something similar via the css prop in that it accepts arrays of styles, but the styles have to be declared/imported, but it's nice to be able to use the pre-defined values for variants.

https://emotion.sh/docs/composition
```
/** @jsx jsx */
import { css, jsx } from '@emotion/core'

const danger = css`
  color: red;
`

const base = css`
  background-color: darkgreen;
  color: turquoise;
`

render(
  <div>
    <div css={base}>This will be turquoise</div>
    <div css={[danger, base]}>
      This will be also be turquoise since the base styles
      overwrite the danger styles.
    </div>
    <div css={[base, danger]}>This will be red</div>
  </div>
)
```

Reply:
Ah, I didn't know that. That's pretty cool that css accepts that. I did update it so that you can do `variants="secondary lg"` too and not have to pass an array.